### PR TITLE
Add ownerReferences in waypoint SA

### PR DESCRIPTION
**Pull request by dddddai**
**Please provide a description of this PR:**
The waypoint SA is not cleaned up after deleting the waypoint, causing resource leak.

### CODE REVIEW

**Comment by howardjohn** [comment]
This is intentional due to a bug in k8s. Will link the issue when I find it

**Comment by hanxiaop** [comment]
This is related to https://github.com/istio/istio/issues/43687#issuecomment-1450253070

**Comment by dddddai** [comment]
Oh thanks for the info
The k8s bug seems to be fixed by https://github.com/kubernetes/kubernetes/pull/117412

**Comment by dddddai** [comment]
The same issue applies to k8s gateway

**Comment by howardjohn** [comment]
Fixed in k8s 1.28 though... could be years before that gets to users

**Comment by dddddai** [comment]
> Fixed in k8s 1.28 though... could be years before that gets to users

Yeah users have to manually delete SA before that..

**Comment by hanxiaop** [comment]
@dddddai @howardjohn do we need to reopen this PR? I'm using Kubernetes 1.29, and the service account still cannot be deleted unless the `ownerReferences` is specified.

**Comment by howardjohn** [comment]
At best we could add a check to add it back if they are on k8s 1.28+ maybe?

File: manifests/charts/istio-control/istio-discovery/files/waypoint.yaml [hunk 1]
```diff
@@ -12,6 +12,14 @@ metadata:
         "gateway.networking.k8s.io/gateway-name" .Name
         "istio.io/gateway-name" .Name
       ) | nindent 4 }}
+  {{- if ge .KubeVersion 128 }}
+  # Safe since 1.28: https://github.com/kubernetes/kubernetes/pull/117412
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:16]: btw, did you verify this? I never really understood how that PR fixed the problem...

**Review by howardjohn — APPROVED** [review]

**Review by hanxiaop — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/files/waypoint.yaml (hunk 1 above)
Comment (dddddai) [reply manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:16]: Yes I did. A pod can be deleted immediately if mounting is failed

**Pull request merged**